lib/split/user.rb

```diff
@@ -83,11 +83,29 @@ def alternative_key_for_experiment(experiment)
     private
 
     def keys_without_experiment(keys, experiment_key)
-      keys.reject { |k| k.match(Regexp.new("^#{experiment_key}(:finished)?$")) || k.match("#{experiment_key}:time_of_assignment") }
+      if experiment_key.include?(':')
+        sub_keys = keys.reject { |k| k == experiment_key }
+        sub_keys.reject do |k|
+          if k.match(Regexp.new("^#{experiment_key}:"))
+            sub_str = k.partition(':').last
+            sub_str.scan(Regexp.new("\\D")).any?
+          else
+            false
+          end
```

Comment:
alternate way to write:
```ruby
return false unless k.match(Regexp.new("^#{experiment_key}:"))
sub_str = k.partition(':').last
sub_str.scan(Regexp.new("\\D")).any?
```

Reply:
I thought about something like this as well. But return `false` will return as the function of the `keys_without_experiment`, instead of just reject. I'm not sure how to bypass it unless having another variable as the return result?

Reply:
return false doesn't work here. I updated with another implementation. 
Thanks!